### PR TITLE
feat!: getIdentitiesContractKeys endpoint

### PR DESCRIPTION
**Pull request by markin-io**
## Issue being fixed or feature implemented

Remove `getIdentities` and `getIdentitiesByPublicKeyHashes` queries, and add `getIdentitiesContractKeys` as a replacement

## What was done?

- Remove queries from proto messages
- Remove queries from drive
- Update JS and Rust clients 
- Added `getIdentitiesContractKeys` and e2e test

## How Has This Been Tested?

- CI

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [ ] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/platform-test-suite/test/functional/platform/proofs.spec.js [hunk 1]
```diff
@@ -233,7 +233,7 @@ describe('Platform', () => {
           });
 
           // eslint-disable-next max-len
-          it('should be able to verify that multiple identities exist with getIdentitiesByPublicKeyHashes', async () => {
+          it.skip('should be able to verify that multiple identities exist with getIdentitiesByPublicKeyHashes', async () => {
```
Comment (shumkov) [inline packages/platform-test-suite/test/functional/platform/proofs.spec.js:236]: Just remove?

**Review by shumkov — CHANGES_REQUESTED** [review]

**Review by shumkov — DISMISSED** [review]
👍

File: packages/rs-drive-abci/src/query/identity_based_queries/identities/mod.rs [hunk 2]
```diff
@@ -1,58 +0,0 @@
...
-            return Ok(QueryValidationResult::new_with_error(
-                QueryError::DecodingError("could not decode identities query".to_string()),
-            ));
-        };
-
-        let feature_version_bounds = &platform_version
-            .drive_abci
-            .query
-            .identity_based_queries
-            .identities;
```
Comment (QuantumExplorer) [inline packages/rs-drive-abci/src/query/identity_based_queries/identities/mod.rs:31]: You forgot to remove the query version.

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

**Review by shumkov — APPROVED** [review]
👍

**Comment by QuantumExplorer** [comment]
Let's wait on the new methods needed before this one is merged in.

File: packages/dapi-grpc/protos/platform/v0/platform.proto [hunk 3]
```diff
@@ -270,6 +236,34 @@ message GetIdentityKeysResponse {
   oneof version { GetIdentityKeysResponseV0 v0 = 1; }
 }
 
+message GetIdentitiesKeysRequest {
+
+  message GetIdentitiesKeysRequestV0 {
+    repeated bytes ids = 1;
```
Comment (QuantumExplorer) [inline packages/dapi-grpc/protos/platform/v0/platform.proto:242]: How is this going to work if you don't have the SearchKey?

File: packages/dapi-grpc/protos/platform/v0/platform.proto [hunk 4]
```diff
@@ -270,6 +236,34 @@ message GetIdentityKeysResponse {
...
+  oneof version { GetIdentitiesKeysRequestV0 v0 = 1; }
+}
+
+message GetIdentitiesKeysResponse {
+
+  message GetIdentitiesKeysResponseV0 {
+    message Keys { repeated bytes keys_bytes = 1; }
+
+    // Map of identity id to keys
+    message IdentitiesKeys { map<string, Keys> entries = 1; };
```
Comment (QuantumExplorer) [inline packages/dapi-grpc/protos/platform/v0/platform.proto:255]: We don't use a map anywhere else and instead give back the values as a vec in the same order it was requested. I'm not super against using a map, I just think that it breaks consistency.

File: packages/dapi-grpc/build.rs [hunk 5]
```diff
@@ -60,13 +58,11 @@ fn configure_platform(mut platform: MappingConfig) -> MappingConfig {
...
 
     //  "GetConsensusParamsResponse" is excluded as this message does not support proofs
-    const VERSIONED_RESPONSES: [&str; 18] = [
+    const VERSIONED_RESPONSES: [&str; 16] = [
         "GetDataContractHistoryResponse",
         "GetDataContractResponse",
         "GetDataContractsResponse",
         "GetDocumentsResponse",
-        "GetIdentitiesByPublicKeyHashesResponse",
-        "GetIdentitiesResponse",
```
Comment (QuantumExplorer) [inline packages/dapi-grpc/build.rs:69]: Same

File: packages/rs-drive-abci/src/query/identity_based_queries/identities_keys/v0/mod.rs [hunk 6]
```diff
@@ -35,49 +34,48 @@ impl<C> Platform<C> {
...
+            //     None,
+            //     &platform_version.drive,
+            // )?;
+            //
+            // GetPartialIdentitiesResponseV0 {
+            //     result: Some(get_partial_identities_response_v0::Result::Proof(
+            //         self.response_proof_v0(platform_state, proof),
+            //     )),
+            //     metadata: Some(self.response_metadata_v0(platform_state)),
+            // }
```
Comment (QuantumExplorer) [inline packages/rs-drive-abci/src/query/identity_based_queries/identities_keys/v0/mod.rs:49]: todo...

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]
The reason why we are not querying all identity information is because we could have 100s of keys on each identity. There is no benefit in getting back all keys for many identities. The goal here is to get specific keys for many identities. For example in Dashpay Dapp if I had 25 contacts, I would want to query the Dashpay contract bound encryption and decryption keys for the 25 contacts I have.

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]
The reason why we are not querying all identity information is because we could have 100s of keys on each identity. There is no benefit in getting back all keys for many identities. The goal here is to get specific keys for many identities. For example in Dashpay Dapp if I had 25 contacts, I would want to query the Dashpay contract bound encryption and decryption keys for the 25 contacts I have.

**Review by QuantumExplorer — DISMISSED** [review]

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**